### PR TITLE
sql: Added destination buffer parameter to `Type.SQL()`

### PR DESCRIPTION
**Pull request by andrew-wm-arthur**

### CODE REVIEW

File: sql/range_tree.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ const (
 
 // RangeColumnExprTree represents a red-black tree over a range column expression. To represent an entire range, each
 // node has both an upper bound and lower bound that represents a single column expression. If the Range has another
-// dimension, then the node will have an inner tree representing the nested dimension ad infinitum. This implicitly
+// dimension, then the node will have an inner tree represeRangeColumnExprTreenting the nested dimension ad infinitum. This implicitly
```
Comment (zachmu) [inline sql/range_tree.go:43]: lol wut

File: sql/settype.go [hunk 2]
```diff
@@ -203,7 +203,7 @@ func (t setType) Promote() Type {
 }
 
 // SQL implements Type interface.
-func (t setType) SQL(v interface{}) (sqltypes.Value, error) {
+func (t setType) SQL(dest []byte, v interface{}) (sqltypes.Value, error) {
```
Comment (zachmu) [inline sql/settype.go:206]: Should get the same treatment as a string type

**Review by zachmu — APPROVED** [review]
LGTM

File: sql/range_tree.go (hunk 1 above)
Comment (andrew-wm-arthur) [reply sql/range_tree.go:43]: w00psd

**Pull request merged**